### ISSUE TITLE
Crunch suddenly doesn't work - Rpc error

### ISSUE THREAD

**Issue by TheCrazyStaker** [turboflakes/crunch#25]
I've been encountering this error lately: 
```
INFO crunch::runtimes::kusama] Inspect and `crunch` unclaimed payout rewards
WARN crunch::runtimes::kusama] batch call indexes [0 : 2]
ERROR crunch::crunch] Subxt error: Rpc error: RPC error
INFOcrunch::crunch] Connected to Kusama network using ws://127.0.0.1:9944 * Substrate node Parity Polkadot v0.9.36-dc25abc712e
INFO crunch::matrix] The 'Crunch Bot' user @......t:matrix.org has been authenticated at matrix.org
INFO crunch::matrix] Messages will be sent to room #Y3J1bmNoL.............
INFOcrunch::runtimes::kusama] Inspect and `crunch` unclaimed payout rewards
WARN  crunch::runtimes::kusama] batch call indexes [0 : 2]
ERROR crunch::crunch] Subxt error: Rpc error: RPC error
```
Some discord member sugested to replace the RPC with 
`CRUNCH_SUBSTRATE_WS_URL=wss://ksm-rpc.stakeworld.io:443
`which I did, but the error is still there and it is quite the same.

I was hoping that reinstalling the node will solve the issue so I reinstalled on a fresh Ubuntu. For some reason I decided to use 20.0.4 LTS, not 22.0.4 like I did before which seems to be a not very inspired choice: now I have another issue: the OpenSSL version is OpenSSL 1.1.1f which looks like it doesn't support crunch.
In the meantime I tried to upgrade my Openssl to 3.1.0, some errors occurred and I just gave up for now as it's Christmas Eve, lol.
I'll be back with a new report after OpenSSL will be upgraded to 3.1.0.

**Comment by paulormart** [comment]
Hi, latest crunch releases [v0.8.1](https://github.com/turboflakes/crunch/releases/tag/v0.8.1) were automatically built under Ubuntu 22.04.1 LTS. I would recommend using the same OS version as well to run `crunch`.

Also could you please provide the full command and flags you are using to run crunch?

Are you running like `/crunch-bot/crunch --config-path /crunch-bot/.env rewards era --seed-path /crunch-bot/.private.seed` or do you use different flags?

**Comment by TheCrazyStaker** [comment]
I always use the latest release, so I'm up to date with 0.8.1. This is my running line
`ExecStart=/crunch-bot/crunch --config-path /crunch-bot/.env rewards   era --seed-path /crunch-bot/.private.seed   --disable-public-matrix-room`
Anyway, after reinstalling the node with Ubuntu 22.04, everything looks back to normal, but the validator is not in the active set yet. I'll be back with new impressions after it will be nominated again.

**Comment by TheCrazyStaker** [comment]
Back into the active set, and the payment failed  again:

```
systemctl status crunch.service
● crunch.service - Kusama Crunch Bot
     Loaded: loaded (/etc/systemd/system/crunch.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2022-12-26 09:16:57 CET; 23h ago
   Main PID: 14582 (crunch)
      Tasks: 28 (limit: 38096)
     Memory: 16.1M
        CPU: 19.521s
     CGroup: /system.slice/crunch.service
             └─14582 /crunch-bot/crunch --config-path /crunch-bot/.env rewards era --seed-path /crunch-bot/.private.seed --disable-public-matrix-room
crunch[14582]: [2022-12-27T07:13:46Z WARN  crunch::runtimes::kusama] batch call indexes [0 : 1]
crunch[14582]: [2022-12-27T07:13:46Z ERROR crunch::crunch] Subxt error: Rpc error: RPC error
crunch[14582]: [2022-12-27T07:18:47Z INFO  crunch::crunch] Connected to Kusama network using ws://127.0.0.1:9944 * Substrate node Parity Polkadot v0.9.36-dc25abc712e
crunch[14582]: [2022-12-27T07:18:47Z INFO  crunch::matrix] The 'Crunch Bot' user @..............has been authenticated at matrix.org
crunch[14582]: [2022-12-27T07:18:47Z INFO  crunch::matrix] Messages will be sent to room #Y3...................=:matrix.org (Private)
crunch[14582]: [2022-12-27T07:18:48Z INFO  crunch::matrix] Crunch Bot (....................) * Matrix bot display name changed
crunch[14582]: [2022-12-27T07:18:48Z INFO  crunch::runtimes::kusama] Inspect and `crunch` unclaimed payout rewards
crunch[14582]: [2022-12-27T07:18:48Z WARN  crunch::runtimes::kusama] Crunch upgrade might be required soon. Local static metadata differs from current chain runtime version.
crunch[14582]: [2022-12-27T07:18:48Z WARN  crunch::runtimes::kusama] batch call indexes [0 : 1]
crunch[14582]: [2022-12-27T07:18:48Z ERROR crunch::crunch] Subxt error: Rpc error: RPC error
```

**Comment by paulormart** [comment]
You can also try to connect to a public RPC node. Example, try to specify the following variables in your `/crunch-bot/.env` file and let me know if the same error occurs:
```
CRUNCH_STASHES=YOUR_VALIDATOR_STASH
CRUNCH_SUBSTRATE_WS_URL=wss://kusama-rpc.polkadot.io:443
CRUNCH_MAXIMUM_PAYOUTS=10
CRUNCH_MAXIMUM_HISTORY_ERAS=16
CRUNCH_EXISTENTIAL_DEPOSIT_FACTOR_WARNING=1000
CRUNCH_MAXIMUM_CALLS=4
CRUNCH_MAXIMUM_PAYOUTS=10
```

**Comment by TheCrazyStaker** [comment]
I did it but the result at the era change a few minutes ago is the same:

```
INFO  crunch::runtimes::kusama] Inspect and `crunch` unclaimed payout rewards
WARN  crunch::runtimes::kusama] Crunch upgrade might be required soon. Local static metadata differs from current chain runtime version.
WARN  crunch::runtimes::kusama] batch call indexes [0 : 1]
ERROR crunch::crunch] Subxt error: Rpc error: RPC error
```

**Comment by validierungcc** [comment]
We have the same problem now

**Comment by validierungcc** [comment]
Okay, it just "fixed itself". The only thing that's different to 10min ago is that I manually did the payouts.

**Comment by TheCrazyStaker** [comment]
> Okay, it just "fixed itself". The only thing that's different to 10min ago is that I manually did the payouts.
Yeah, but wait until next era.
If I perform the payments manually, then the crunch logs and status looks fine, but when era changes, it won't perform the payment and the error comes back until I perform the payments manually again.

**Comment by TheCrazyStaker** [comment]
as  you can see, this evening at 20.59 I perform the payment manually. until then, crunch retried every 5 minutes to do it but it couldn't.  now everything looks fine but this will be until 2:27 AM when it will start throwing that error again

![image](https://user-images.githubusercontent.com/98461034/209866687-5f1a3fcf-2d4d-46a4-8535-425652bca0b9.png)

**Comment by paulormart** [comment]
Could the reason be the signer account that doesn’t have enough funds to pay for the transaction? Try top-up that account with a small amount and see if it fixes the issue.

**Comment by TheCrazyStaker** [comment]
I can't believe it worked. I mean I can't believe because a few weeks ago, I had this issue, but back then it was clearly stated in the log error: "funds not enough " or something like this. This time this "funds error" was not stated so I got that out of the question. Doh... happy ending (I hope)!
Thank you Paulo for your assistance!

**Comment by TheCrazyStaker** [comment]
So yes, now that the era changed I can confirm: everything works properly. Thanks!

**Comment by paulormart** [comment]
Ok thanks for the feedback. Glad it's working again, I’ll try to review that error log to be more explicit.

**Comment by validierungcc** [comment]
Works for me too now. the suggested fee was slightly higher than what I had on the account. the real fee was 10 times lower, though.

**Comment by paulormart** [comment]
error readability fixed in v0.10.0
